### ISSUE TITLE
Does not work in paths containing spaces

### ISSUE THREAD

**Issue by Kwarf** [ryanmjacobs/c#26]
The script does not work on files with paths containing whitespaces.

```
[jimmy@toolbox my folder]$ pwd
/home/jimmy/my folder
[jimmy@toolbox my folder]$ ./foo.c 
cc: error: folder: No such file or directory
[jimmy@toolbox my folder]$ c foo.c
cc: error: folder: No such file or directory
[jimmy@toolbox my folder]$ c "/home/jimmy/my folder/foo.c"
cc: error: /home/jimmy/my: No such file or directory
cc: error: folder/foo.c: No such file or directory
cc: error: folder: No such file or directory
cc: fatal error: no input files
compilation terminated.
```

**Comment by ryanmjacobs** [comment]
@Kwarf I've made a fix on a separate branch. Would you mind trying again? Just pull again and checkout `args-with-spaces`.

**Comment by Kwarf** [comment]
It seems to be working when using the shebang, however it breaks passing compiler arguments from the command line.
```
[jimmy@toolbox ~]$ c "foo.c -std=c11 -lm --"
cc: error: foo.c -std=c11 -lm --: No such file or directory
cc: fatal error: no input files
compilation terminated.
```
The above command works on the version in master (as long as there's no spaces in the path of course).

**Comment by ryanmjacobs** [comment]
Hmm, I get the same thing. I'll have to try something else later.

On a side note, the only time `--` needs to be used is on the shebang line. The script takes the argument that immediately follows it and uses it for `argv[0]`. When you execute a script using the shebang, it runs like so: `<interpreter> [flags | args | etc...] <filename>`. So I decided to use `--` to separate the compiler flags and files from the main filename, i.e. `<interpreter> [flags | args | etc...] -- <filename>`.

So you could trying again without the `--`. Hopefully, that helps a little.

**Comment by ryanmjacobs** [comment]
I'm not quite sure how to fix this, so I'll just leave it open until someone else or myself figures it out...

**Comment by frebib** [comment]
Just remembered this was a thing. Came to use it today and immediately ran into this issue.

**Comment by hmocheimech** [comment]
use 'ls- a' or 'ls -la ' or the' find' command to locate the **file full path**, I also recommend that you use the  'help' command since there's a lot of extension for each command.

**Comment by ryanmjacobs** [comment]
Should be fixed as of v0.11

Reopen this issue if there's some edge case I missed.
